test/common_utils.py

```diff
@@ -14,3 +20,135 @@ def get_tmp_dir(src=None, **kwargs):
         yield tmp_dir
     finally:
         shutil.rmtree(tmp_dir)
+
+ACCEPT = os.getenv('EXPECTTEST_ACCEPT')
+
+parser = argparse.ArgumentParser(add_help=False)
+parser.add_argument('--accept', action='store_true')
+args, remaining = parser.parse_known_args()
+if not ACCEPT:
+    ACCEPT = args.accept
+for i, arg in enumerate(sys.argv):
+    if arg == '--accept':
+        del sys.argv[i]
+        break
+
+
+class MapNestedTensorObjectImpl(object):
+    def __init__(self, tensor_map_fn):
+        self.tensor_map_fn = tensor_map_fn
+
+    def __call__(self, object):
+        if isinstance(object, torch.Tensor):
+            return self.tensor_map_fn(object)
+
+        elif isinstance(object, dict):
+            mapped_dict = {}
+            for key, value in object.items():
+                mapped_dict[self(key)] = self(value)
+            return mapped_dict
+
+        elif isinstance(object, (list, tuple)):
+            mapped_iter = []
+            for iter in object:
+                mapped_iter.append(self(iter))
+            return mapped_iter if not isinstance(object, tuple) else tuple(mapped_iter)
+
+        else:
+            return object
+
+
+def map_nested_tensor_object(object, tensor_map_fn):
+    impl = MapNestedTensorObjectImpl(tensor_map_fn)
+    return impl(object)
+
+
+# adapted from TestCase in torch/test/common_utils
```

Comment:
could you point out the main changes / simplifications compared to what we have in `common_utils`, just for awareness? Did you just removed functionality, or did you change a few things as well?